### PR TITLE
Adds a small recovery window to variable length megafauna attacks for more predictable melee resolution

### PR DESCRIPTION
**Pull request by Ergovisavi**
Adds a small window to variable length attacks on megafauna where they can't melee. This basically is just the Swoop (Ash Drake) and Charge (Bubblegum) attacks. There are cases where it's impossible to tell if it's safe to attack or not because the Life() ticks that determine when the megafauna will swipe at you.

L....L....L....L....L....
(L) is a Life() tick, where the mob will melee you, you're safe to melee on any of the .'s, then we add in variable length attacks(^)

L....L....L....L....L....
.^....^.............^

Now you can't tell when it's safe to attack and may get slapped in melee even though you were "Playing correctly", which feels random and arbitrary. Now it's more predictable. This change doesn't really affect anyone shooting megafauna with the PKA, but will be useful for melee attacks.

:cl:
tweak: Added a recovery window after some variable length megafauna attacks
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 1]
```diff
@@ -121,7 +124,8 @@
 		if(3)
 			adjustBruteLoss(50)
 
-
+/mob/living/simple_animal/hostile/megafauna/proc/SetRecoveryTime(var/buffer_time)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: No ```var/```

This keeps happening so often lately I'm really tempted to write a script to auto-fail travis.

**Comment by ChangelingRain** [comment]
Honestly melee attacks from mobs need to be more predictable in general

**Comment by Ergovisavi** [comment]
I agree, but we're A) On a tile based system, B) We move at 99 billion MPH and C) We're gonna have to put attack animations on everything, D) Most use cases that don't involve mining can be bypassed by the throwing mechanic

It's not an easy thing to fix without making everything entirely irrelevant

**Comment by Expletive** [comment]
@Ergovisavi  The only one insane enough to try and melee a megafauna is you though

File: code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm [hunk 2]
```diff
@@ -258,6 +258,7 @@ Difficulty: Medium
 	density = TRUE
 	sleep(1)
 	swooping &= ~SWOOP_DAMAGEABLE
+	SetRecoveryTime(5)
```
Comment (lzimann) [inline code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm]: These could use a define

**Comment by lzimann** [comment]
@Ergovisavi defines please!

**Comment by ChangelingRain** [comment]
I don't think this actually really fixes the problem, just obscures it a bit; I'll explain why in the morning when I can set up a little chart dealie like in the PR body.

**Comment by Ergovisavi** [comment]
Obfuscating the problem is fine as long as it feels good for players. If you have a better solution we can easily implement, I'd love to hear it.

**Comment by ChangelingRain** [comment]
The thing is, this doesn't feel good for players since it's not actually changing when the next attack will come out(ie, in-between moves instead of after moves), just keeping it from immediately attacking after an attack. A variable-length attack still changes when it attacks, so this just obscures it and doesn't majorly improve the experience for players.

Also it turns out those little chart dealies are hard to do wow

**Comment by Ergovisavi** [comment]
That's not entirely accurate, though. All the megafauna attacks us OpenFire(), there's a bare minimum amount of time between the big megafauna attacks, based on their ranged_attack_cooldown, and that isn't changed between variable length attacks except in cases where one of them extends into the other, which is basically pretty impossible?

**Comment by KorPhaeron** [comment]
Ergo is way smarter than me so you can merge his thing whenever @lzimann

**Comment by Ergovisavi** [comment]
I hate git. Somehow I managed to commit some weirdass ghost file. Halp.

**Pull request merged**